src/modules/content-sqlite/content-sqlite.c

```diff
@@ -353,18 +356,33 @@ void checkpoint_get_cb (flux_t *h,
         errno = ENOENT;
         goto error;
     }
+    s = (char *)sqlite3_column_text (ctx->checkpt_get_stmt, 0);
+    if (!(o = json_loads (s, 0, NULL))) {
+        /* version 0 checkpoint is a blobref string, used before v0.36
+         * reply with updated object representation.
+         */
+        if (!(o = json_pack ("{s:i s:s s:f}",
+                             "version", 0,
+                             "rootref", s,
+                             "timestamp", 0.))) {
+            errno = EINVAL;
+            goto error;
+        }
+    }
```

Comment:
In the event that data becomes corrupted in the sqlite database, it may be good to have a little tighter error checking here and not return json-encoded garbage to the caller.   Maybe something like
```c
if (!(o = json_loads (s, 0, &error))) {
   if (blobref_validate (s) < 0) {
      errstr = error.text;
      errno = EINVAL;
      goto error;
   }
   /* assume "version 0" if value is a bare blobref and return it in a json envelope */
   ...
}